File: test-automation-core/src/main/java/com/nortal/test/core/testng/SpringContextHolder.java
package com.nortal.test.core.testng;

import org.springframework.context.ConfigurableApplicationContext;

/**
 * Abstracts the context holder from this package
 * 
 * @author VJose6
 *
 */
public interface SpringContextHolder {
	
	ConfigurableApplicationContext getApplicationContext();

	void beforeTestClass() throws Exception;

	void afterTestClass() throws Exception;

}

File: test-automation-core-old/src/main/java/com/nortal/test/core/testng/SpringContextHolder.kt
package com.nortal.test.core.testng

import org.springframework.context.ConfigurableApplicationContext
import java.lang.Exception

/**
 * Abstracts the context holder from this package
 *
 * @author VJose6
 */
interface SpringContextHolder {
    val applicationContext: ConfigurableApplicationContext?

    @Throws(Exception::class)
    fun beforeTestClass()

    @Throws(Exception::class)
    fun afterTestClass()
}